pkg/operation/botanist/secrets.go

```diff
@@ -1166,10 +1167,10 @@ func dnsNamesForService(name, namespace string) []string {
 
 func dnsNamesForEtcd(namespace string) []string {
```

Comment:
I thought we agreed on the fact that Gardener does not generate any certificates for etcd anymore. I think this whole code related to etcd here can be deleted.

Reply:
I have to check on how this would affect backward compatibility on existing shoots, though.

Reply:
We have decided to add a new secret with etcd-'role'-local endpoint to the Alternate Names in the server certificate. We shall redact the previous etcd-server-tls certificate in the subsequent gardener release.